### PR TITLE
Make `Config` optimizations and `HTTPMethod` consts

### PR DESCRIPTION
**Pull request by michplunkett**
## Fixes issue
 https://github.com/lucyparsons/OpenOversight/issues/954

## Description of Changes
Added inheritance to the `Config` objects, removed unnecessary functions from the `Config` classes, and removed the custom constants and replaced them with `HTTPMethod` constants.

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: dockerfiles/web/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ARG TRAVIS_PYTHON_VERSION
 ARG DOCKER_BUILD_ENV
-FROM python:${TRAVIS_PYTHON_VERSION:-3.8}-buster
+FROM python:${TRAVIS_PYTHON_VERSION:-3.11}-buster
```
Comment (michplunkett) [inline dockerfiles/web/Dockerfile:3]: Ideally this change is merged with this PR, but I thought it'd be good to test these changes with 3.11 in the mean time: https://github.com/lucyparsons/OpenOversight/pull/962

**Comment by michplunkett** [comment]
The linked PR is likely gonna cause some merge difficulties: [Link](https://github.com/lucyparsons/OpenOversight/pull/944)

File: OpenOversight/app/utils/constants.py [hunk 2]
```diff
@@ -4,11 +4,5 @@
 # Encoding constants
 ENCODING_UTF_8 = "utf-8"
 
-# HTTP Method constants
-# TODO: Remove these constants and use HTTPMethod in http package when we
-#  migrate to version 3.11
-HTTP_METHOD_GET = "GET"
-HTTP_METHOD_POST = "POST"
-
```
Comment (michplunkett) [inline OpenOversight/app/utils/constants.py:12]: RIP.

**Review by abandoned-prototype — APPROVED** [review]
Looks good. Should be fine to merge with that one change.

File: OpenOversight/tests/conftest.py [hunk 3]
```diff
@@ -13,20 +13,21 @@
...
 from OpenOversight.app import create_app, models
+from OpenOversight.app.config import TestingConfig
 from OpenOversight.app.models import Job, Officer, Unit
 from OpenOversight.app.models import db as _db
 from OpenOversight.app.utils.constants import ENCODING_UTF_8
 from OpenOversight.app.utils.general import merge_dicts
 from OpenOversight.tests.routes.route_helpers import ADMIN_EMAIL, ADMIN_PASSWORD
 
 
+CONFIG = TestingConfig()
```
[the remark this answers is not in the document]
Comment (michplunkett) [reply OpenOversight/tests/conftest.py:30]: That works for me! Let me make that change real quick.

[the remark this answers is not in the document]
Comment (michplunkett) [reply OpenOversight/tests/conftest.py]: Done!

**Pull request merged**